src/haxelib/api/Installer.hx

```diff
@@ -0,0 +1,852 @@
+
+		scope.setVersion(library, version);
+
+		userInterface.log('  Current version is now $version');
+		userInterface.log("Done");
+
+		handleDependencies(library, version);
+	}
+
+	/**
+		Installs `version` of `library` from the haxelib server.
+
+		If `forceSet` is set to true and the installer is running with
+		a global scope, the new version is always set as the current one.
+	 **/
+	public function installFromHaxelib(library:ProjectName, version:SemVer, forceSet:Bool = false) {
+		final info = Connection.getInfo(library);
+		// get correct name capitalization so that the logged output is correct
+		final library = ProjectName.ofString(info.name);
+
+		final versions = [for (v in info.versions) v.name];
+
+		if (versions.length == 0)
+			throw new InstallationException('The library $library has not yet released a version');
+		if (!versions.contains(version))
+			throw new InstallationException('No such version $version for library $library');
+
+		downloadAndInstall(library, version);
+
+		if (scope.isLocal || version == getLatest(versions) || !scope.isLibraryInstalled(library) || forceSet) {
+			scope.setVersion(library, version);
+			userInterface.log('  Current version is now $version');
+		}
+		userInterface.log("Done");
+
+		handleDependencies(library, version);
+	}
+
+	/**
+		Installs `library` from a git or hg repository (specified by `id`)
+
+		`vcsData` contains information on the source repository
+		and the requested state.
+	**/
+	public function installVcsLibrary(library:ProjectName, id:VcsID, vcsData:VcsData) {
+		installVcs(library, id, vcsData);
+
+		library = getName(library, id, vcsData.subDir);
+
+		scope.setVcsVersion(library, id, vcsData);
+
+		if (vcsData.subDir != null) {
+			final path = scope.getPath(library);
+			userInterface.log('  Development directory set to $path');
+		} else {
+			userInterface.log('  Current version is now $id');
+		}
+		userInterface.log("Done");
+
+		handleDependenciesVcs(library, id, vcsData.subDir);
+	}
+
+	/**
+		Updates `library` to the newest version on the haxelib server,
+		or pull latest changes with git or hg.
+	**/
+	public function update(library:ProjectName) {
+		try {
+			updateIfNeeded(library);
+		} catch (e:AlreadyUpToDate) {
+			userInterface.log(e.toString());
+		} catch (e:VcsUpdateCancelled) {
+			return;
+		}
+	}
+
+	/**
+		Updates all libraries in the scope.
+
+		If a library update fails, it is skipped.
+	**/
+	public function updateAll():Void {
+		final libraries = scope.getLibraryNames();
+		var updated = false;
+
+		for (library in libraries) {
+			userInterface.log('Checking $library');
+			try {
+				updateIfNeeded(library);
+				updated = true;
+			} catch(e:AlreadyUpToDate) {
+				continue;
+			} catch (e:VcsUpdateCancelled) {
+				continue;
+			} catch(e) {
+				userInterface.log("Failed to update: " + e.toString());
+				userInterface.log(e.stack.toString(), Debug);
+			}
+		}
+
+		if (updated)
+			userInterface.log("All libraries are now up-to-date");
```

Comment:
This seems to be a potential lie, `updated` is set to true as soon as _one_ update succeeds in the loop above.